### ISSUE TITLE
build failed

### ISSUE THREAD

**Issue by fiag** [airtrack/stunnel#6]
```
   Compiling tide v0.14.0
error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/listener/failover_listener.rs:90:21
   |
90 | /                     crate::log::info!("unable to listen", {
91 | |                         listener: listener.to_string(),
92 | |                         error: e.to_string()
93 | |                     });
   | |_______________________^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/log/middleware.rs:43:9
   |
43 | /         log::info!("<-- Request received", {
44 | |             method: method,
45 | |             path: path,
46 | |         });
   | |___________^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/log/middleware.rs:52:17
   |
52 | /                 log::error!("Internal error --> Response sent", {
53 | |                     message: format!("{:?}", error),
54 | |                     error_type: error.type_name(),
55 | |                     method: method,
...  |
58 | |                     duration: format!("{:?}", start.elapsed()),
59 | |                 });
   | |___________________^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/log/middleware.rs:61:17
   |
61 | /                 log::error!("Internal error --> Response sent", {
62 | |                     method: method,
63 | |                     path: path,
64 | |                     status: format!("{} - {}", status as u16, status.canonical_reason()),
65 | |                     duration: format!("{:?}", start.elapsed()),
66 | |                 });
   | |___________________^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/log/middleware.rs:70:17
   |
70 | /                 log::warn!("Client error --> Response sent", {
71 | |                     message: format!("{:?}", error),
72 | |                     error_type: error.type_name(),
73 | |                     method: method,
...  |
76 | |                     duration: format!("{:?}", start.elapsed()),
77 | |                 });
   | |___________________^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/log/middleware.rs:79:17
   |
79 | /                 log::warn!("Client error --> Response sent", {
80 | |                     method: method,
81 | |                     path: path,
82 | |                     status: format!("{} - {}", status as u16, status.canonical_reason()),
83 | |                     duration: format!("{:?}", start.elapsed()),
84 | |                 });
   | |___________________^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/log/middleware.rs:87:13
   |
87 | /             log::info!("--> Response sent", {
88 | |                 method: method,
89 | |                 path: path,
90 | |                 status: format!("{} - {}", status as u16, status.canonical_reason()),
91 | |                 duration: format!("{:?}", start.elapsed()),
92 | |             });
   | |_______________^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/sessions/middleware.rs:99:17
   |
99 |                 crate::log::error!("unable to destroy session", { error: e.to_string() });
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/listener/tcp_listener.rs:66:13
   |
66 |             log::error!("async-h1 error", { error: error.to_string() });
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/listener/unix_listener.rs:79:13
   |
79 |             log::error!("async-h1 error", { error: error.to_string() });
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `std::string::String: ToValue` is not satisfied
  --> ./tide-0.14.0/src/log/mod.rs:40:5
   |
40 |     crate::log::info!("Logger started", { level: format!("{}", level) });
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `ToValue` is not implemented for `std::string::String`
   |
   = note: required for the cast to the object type `dyn ToValue`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 11 previous errors

For more information about this error, try `rustc --explain E0277`.
error: could not compile `tide`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

rustc 1.51.0-nightly (d1aed50ab 2021-01-26)

**Comment by fiag** [comment]
应该是tide的问题 https://github.com/http-rs/tide/issues/787

**Comment by airtrack** [comment]
Tide fixes this issue in release [v0.15.1](https://github.com/http-rs/tide/releases/tag/v0.15.1).
